test/xds/xds_server_serving_mode_test.go

```diff
@@ -112,9 +112,18 @@ func (s) TestServerSideXDS_RedundantUpdateSuppression(t *testing.T) {
 	// suppressed, server will recycle client connections.
 	errCh := make(chan error, 1)
 	go func() {
-		if cc.WaitForStateChange(ctx, connectivity.Ready) {
-			errCh <- fmt.Errorf("unexpected connectivity state change {%s --> %s} on the client connection", connectivity.Ready, cc.GetState())
-			return
+		prev := connectivity.Ready // We know we are READY since we just did an RPC.
+		for {
+			curr := cc.GetState()
```

Comment:
qq: what happens if the context gets cancelled when you are here. Will the WaitForStateChange call later still return false? It reads the ctx.Done() channel which has already been closed, will this Done channel still send a zero value that signifies it's been closed?

Reply:
"Successive calls to Done return the same value."

"A receive operation on a [closed](https://go.dev/ref/spec#Close) channel can always proceed immediately, yielding the element type's [zero value](https://go.dev/ref/spec#The_zero_value) after any previously sent values have been received."

I think these statements from godoc mean the answer to my question is yes, but just making sure.